pandas/core/generic.py

```diff
@@ -5410,19 +5405,15 @@ def __setattr__(self, name: str, value) -> None:
 
         # if this fails, go on to more involved attribute setting
         # (note that this matches __getattr__, above).
-        if name in self._internal_names_set:
-            object.__setattr__(self, name, value)
-        elif name in self._metadata:
+        if name in self._internal_names_set or name in self._metadata:
             object.__setattr__(self, name, value)
         else:
             try:
                 existing = getattr(self, name)
-                if isinstance(existing, Index):
+                if isinstance(existing, Index) or name not in self._info_axis:
                     object.__setattr__(self, name, value)
-                elif name in self._info_axis:
-                    self[name] = value
                 else:
-                    object.__setattr__(self, name, value)
+                    self[name] = value
```

Comment:
can you check the logic that was changed here (IIRC this is for performance but the change might be good)

Reply:
I am not sure about this change on L5509. pls revert and can do in a separate PR when checking perf. This causes recursion (which might be fine, though need to confirm)